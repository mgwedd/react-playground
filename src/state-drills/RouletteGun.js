import React, { Component } from 'react';

export default class RouletteGun extends Component {
    
    state = {
        chamber: null, 
        spinningTheChamber: false
    }

    handleTrigger = () => {
        this.setState({spinningTheChamber: true});
        this.timeout = setTimeout(() => {
            const randomChamber = Math.ceil(Math.random() * 8);
            this.setState({chamber: randomChamber, spinningTheChamber: false})
        }, 2000);
    }

    componentWillUnmount() {
        clearTimeout(this.timeout);
    }

    renderDisplay = () => {
       const { spinningTheChamber, chamber } = this.state;
       const { bulletInChamber } = this.props;
        if (spinningTheChamber) {
            return 'spinning the chamber and pulling the trigger! ...'
          } else if (chamber === bulletInChamber) {
            return 'BANG!!!!!'
          } else {
            return 'you\'re safe!'
          }
    }
    render() {
        return (
            <div>
                <button onClick={this.handleTrigger}>Pull the trigger!</button>
                <p>{this.renderDisplay()}</p>
            </div>
        );
    }
}

RouletteGun.defaultProps = { bulletInChamber: 8 };
