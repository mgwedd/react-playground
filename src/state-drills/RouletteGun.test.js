import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RouletteGun from './RouletteGun';

describe('RouletteGun Component', () => {

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RouletteGun />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
    const tree = renderer
        .create(<RouletteGun />)
        .toJSON();
    expect(tree).toMatchSnapshot();  
    });
    
});