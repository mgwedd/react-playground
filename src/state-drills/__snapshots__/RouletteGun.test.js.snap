// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouletteGun Component renders the UI as expected 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Pull the trigger!
  </button>
  <p>
    you're safe!
  </p>
</div>
`;
